pallets/asset-depository/src/traits.rs

```diff
@@ -0,0 +1,22 @@
+// Copyright 2021 ChainSafe Systems
+// SPDX-License-Identifier: LGPL-3.0-only
+
+use frame_support::sp_runtime::DispatchResult;
+
+/// An abstraction over the multiple balances for different assets
+pub trait MultiAssetDepository<AssetId, AccountId, Balance> {
+    /// The total amount of the given asset currently held
+    fn aggregated_balance(asset_id: &AssetId) -> Balance;
+
+    /// The total balance of an asset of a user
+    fn total_balance(asset_id: &AssetId, who: &AccountId) -> Balance;
```

Comment:
So, if we want to get `reversed_balance` in the future, we prefer to use `who.total_balance().saturating_sub(who.available_balance())`?

Reply:
Eventually we can add a `LockableAssetDepository` that will make use of `reserve`. My guess at this moment is, that we could use something like this when need to temporarily lock something because we're waiting on the price oracle or if we need to disable deposit/withdraw for certain assets momentarily.